server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -219,7 +219,7 @@ void execute() {
             logger.trace("starting async refresh");
 
             try (var ignoredRefs = fetchRefs) {
-                maybeFetchIndicesStats(diskThresholdEnabled || writeLoadConstraintEnabled == WriteLoadDeciderStatus.ENABLED);
+                maybeFetchIndicesStats(diskThresholdEnabled || writeLoadConstraintEnabled.atLeastLowThresholdEnabled());
```

Comment:
Nit: alternatively we could have a method `WriteLoadDeciderStatus#requiresShardLevelWriteLoads()` (and `requiresNodeLevelWriteLoads()`) which would return `true` for `LOW_THRESHOLD_ONLY` and `ENABLED` but `false` for `DISABLED`.

It would read nicer if the `writeLoadConstraintEnabled` field was called `writeLoadDeciderStatus` if we went that way.

Don't feel strongly about this naming thing though.